### PR TITLE
sh - delete image

### LINKED ISSUE #8: Delete Image
#### As a user I want to delete an image.

__Story__:
Our users are awesome at adding images to our web service. Some of our users
add dozens in a single session. Occasionally, however, some users add the wrong
image, and would like a way to remove those images. Please provide our users
with the ability to delete images. Note that at this point we trust our users
not to abuse this feature and delete more images than they should.

__Acceptance criteria__:
- [x] The user is prompted for confirmation prior to deletion. (Use javascript
  confirm() dialog)
- [x] Deleted images are removed from the homepage.
- [ ] A flow test using page objects exists for the flow to create and
  subsequently delete an image.

__Dependencies__:
- Image Index

__Discussion topics__:
- [ ] What makes a good page object interface?

__Reference material__:
- https://developer.mozilla.org/en-US/docs/Web/API/Window/confirm
- https://github.com/appfolio/ae_page_objects
- http://martinfowler.com/bliki/PageObject.html
- https://sites.google.com/a/appfolio.com/eng/new-hire-resources/engineering-academy/ropes/image-sharing-project/flow-tests-with-page-objects
- https://github.com/SeleniumHQ/selenium/wiki/Ruby-Bindings#javascript-dialogs

__Flow Test Info__:

Now that we have gotten some experience writing flow tests with Capybara,
we've noticed that these tests a) can be brittle in that changing selectors
requires updating many tests and b) use a low-level interface that can make
reading and writing flow tests more cumbersome than we'd like.

Fortunately, there is a better way: using page objects. With page objects,
we can create objects to encapsulate pages, elements, and actions that can be
taken in our application. We can design an interface for interacting with our
application that maps directly to how our users would think about interacting
with the application. This interface makes the tests easier to read, write,
and modify, and reduces the possibility of writing flaky or invalid tests.

Working on this story will be a bit different than your typical stories. We've
created a starting point for you in the form of a branch in your repo called
`flowTestsWithPageObjects`. This branch contains flow tests for adding images,
deleting images, and viewing images associated with a tag (in the
`images_crud_test.rb` file). These tests are fully written, but the page
objects they use are not. Your job is to fill in the page objects to make the
tests pass against your application.

The first step is to rebase the `flowTestsWithPageObjects` branch on top of
your updated master branch. Note that if you already have an
`images_crud_test.rb` file, there will be a merge conflict, so you may want to
rename your file first, rebase, make the tests on that branch pass, and then
migrate any other tests that were in your file to the page object style before
deleting your old test file.

Finally, there is a video to help you get started. You should watch this video
and also take a look at the [ae_page_objects readme](https://github.com/appfolio/ae_page_objects)
which will be an extremely helpful resource as you learn to write page objects. Both are linked in the
reference material below.

__Important Notes__:

Upgrade your Firefox to the latest version.

When dealing with javascript confirm() dialog in the flow test, this [reference](https://github.com/SeleniumHQ/selenium/wiki/Ruby-Bindings#javascript-dialogs)
will be helpful.

### PR DESCRIPTION
**Pull request by seanhelvey**
stoked to get the page objects tests working!

Closes #8

### CODE REVIEW

File: test/fixtures/images.yml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 # Read about fixtures at http://api.rubyonrails.org/classes/ActiveRecord/FixtureSet.html
 
 one:
-  url: MyString
+  url: 'https://www.w3schools.com/w3css/img_lights.jpg'
 
 two:
-  url: MyString
+  url: 'https://images.unsplash.com/photo-1509233725247-49e657c54213?ixlib=rb-1.2.1'
```
Comment (grahamotte) [inline test/fixtures/images.yml]: We usually don't use fixtures in favor exemplars

File: test/controllers/images_controller_test.rb [hunk 2]
```diff
@@ -64,4 +64,11 @@ def test_index__with_params
     assert_response :success
     assert_select 'img', 1
   end
+
+  def test_destroy
```
Comment (grahamotte) [inline test/controllers/images_controller_test.rb]: Prob worth having a test that an image with tags also remove all the tag associations correctly

File: test/fixtures/images.yml (hunk 1 above)
Comment (seanhelvey) [reply test/fixtures/images.yml]: Done

File: test/controllers/images_controller_test.rb (hunk 2 above)
Comment (seanhelvey) [reply test/controllers/images_controller_test.rb]: Added

File: test/test_helper.rb [hunk 3]
```diff
@@ -8,5 +8,8 @@ class TestCase
     fixtures :all
 
     # Add more helper methods to be used by all tests here...
+    def create_image!(url = 'https://www.w3schools.com/w3css/img_lights.jpg', tag_list = 'lights')
```
Comment (grahamotte) [inline test/test_helper.rb]: It's a good pattern in ruby to use keyword args pretty much whenever possible, these positional arguments will break pretty fast

File: test/fixtures/images.yml (hunk 1 above)
Comment (grahamotte) [reply test/fixtures/images.yml]: These look to still be here, was that intentional?

File: test/test_helper.rb (hunk 3 above)
Comment (seanhelvey) [reply test/test_helper.rb]: Thanks for pointing that out! Fixed.

File: test/fixtures/images.yml (hunk 1 above)
Comment (seanhelvey) [reply test/fixtures/images.yml]: Nope! Thank you. Removed.

**Pull request merged**